stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -704,21 +717,32 @@
 ;; See `get-signer-key-message-hash` for details on the message hash.
 ;;
 ;; Note that `reward-cycle` corresponds to the _current_ reward cycle,
-;; not the reward cycle at which the delegation will start.
-;; The public key is recovered from the signature and compared to `signer-key`.
+;; when used with `stack-stx` and `stack-extend`.
+;; When `signer-sig` is present, the public key is recovered from the signature 
+;; and compared to `signer-key`.
+;; 
+;; If `signer-sig` is `none`, the function verifies that an authorization was previously
+;; added for this key.
 (define-read-only (verify-signer-key-sig (pox-addr { version: (buff 1), hashbytes: (buff 32) })
                                          (reward-cycle uint)
                                          (topic (string-ascii 12))
                                          (period uint)
-                                         (signer-sig (buff 65))
+                                         (signer-sig-opt (optional (buff 65)))
```

Comment:
I believe this & only this parameter was updated to include '-opt,' mistake?

Reply:
Yeah you're right, but I did it so that I could use the variable `signer-sig` in the `match` branch. I can update this though to just use `sig` in the match and then we can keep consistent names.

Reply:
I personally prefer `-opt` as a suffix to indicate in the code body that it's an `optional`, but I don't feel it so strongly that it's a blocker.